packages/devextreme/js/__internal/ui/chat/messagelist.ts

```diff
@@ -393,34 +392,56 @@ class MessageList extends Widget<Properties> {
   }
 
   _renderMessage(message: Message): void {
-    const { author, timestamp } = message;
-
-    const lastMessageGroup = this._getLastMessageGroup();
+    const { timestamp } = message;
     const shouldCreateDayHeader = this._shouldAddDayHeader(timestamp);
 
-    if (lastMessageGroup) {
-      const { items } = lastMessageGroup.option();
-      const lastMessageGroupItem = items[items.length - 1];
-      const lastMessageGroupUserId = lastMessageGroupItem.author?.id;
-      const isTimeoutExceeded = this._isTimeoutExceeded(lastMessageGroupItem, message);
-
-      if (author?.id === lastMessageGroupUserId && !isTimeoutExceeded && !shouldCreateDayHeader) {
-        lastMessageGroup.renderMessage(message);
+    if (shouldCreateDayHeader) {
+      this._createDayHeader(timestamp);
+      this._renderMessageIntoGroup(message);
+      return;
+    }
 
-        this._scrollDownContent();
+    const lastMessageGroup = this._getLastMessageGroup();
 
-        return;
-      }
+    if (!lastMessageGroup) {
+      this._renderMessageIntoGroup(message);
+      return;
     }
 
-    if (shouldCreateDayHeader) {
-      this._createDayHeader(timestamp);
+    const lastMessageGroupMessage = this._getLastMessageGroupItem(lastMessageGroup);
+    const isTimeoutExceeded = this._isTimeoutExceeded(lastMessageGroupMessage, message);
+
+    if (this._isSameAuthor(message, lastMessageGroupMessage) && !isTimeoutExceeded) {
+      this._renderMessageIntoGroup(message, lastMessageGroup);
+      return;
     }
 
-    this._createMessageGroupComponent([message], author?.id);
-    this._setLastMessageGroupClasses();
+    this._renderMessageIntoGroup(message);
+  }
 
-    this._scrollDownContent();
+  _getLastMessageGroupItem(lastMessageGroup: MessageGroup): Message {
+    const { items } = lastMessageGroup.option();
+
+    return items[items.length - 1];
```

Comment:
[minor]
```suggestion
    return items.at(-1);
```

Reply:
Discussed in a voice, decided to decline this suggestion because ts error arise after this